presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorMetadata.java

```diff
@@ -223,6 +226,14 @@ default ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTabl
         return emptyMap();
     }
 
+    /**
+     * Returns a (table name, Optional.empty()) entry if the table is redirected. See {@link #redirectTable}'s doc for details.
+     */
+    default Map<SchemaTableName, Optional<List<ColumnMetadata>>> listTableColumnsWithRedirection(ConnectorSession session, SchemaTablePrefix prefix)
```

Comment:
If i implement this method, do i still need to implement `listTableColumns`?

same for other `WithRedirection` methods

Reply:
Nope. I'll add to the doc

Reply:
Now i understand (https://github.com/prestosql/presto/pull/5160#discussion_r517061612).

Thus it's valid to implement `listTableColumns`, but the method should not be called anymore.
I wish there was a way to express that in Java, but there doesn't seem to be.